File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        lowest_price = prices[0]

        for i in range(1, len(prices)):
            max_profit = max(max_profit, prices[i] - lowest_price)
            lowest_price = min(lowest_price, prices[i])
        
        return max_profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.swift
class Solution {
    func maxProfit(_ prices: [Int]) -> Int {
        var max_profit = 0
        var min_price = prices[0]

        for price in prices {
            max_profit = max(max_profit, price - min_price)
            min_price = min(min_price, price)
        }

        return max_profit
        
    }
}